packages/data-provider/src/utils/auditUtils.ts

```diff
@@ -1,19 +1,31 @@
 import * as _ from 'lodash-es';
+import { z } from 'zod';
 
-import { AUDIT_ACTION, AuditAction, AuditDataRepository, AuditDataResponse } from './types.js';
+import { AUDIT_ACTION, AuditAction, AuditDataResponse, AuditRepositoryRecord } from './types.js';
+
+/**
+ * Convert Audit Action enum into an array of strings in uppercase to facilitate validation
+ */
+const upperCaseAuditActions: string[] = Object.values(AUDIT_ACTION).map((value) => value.toUpperCase());
+
+/**
+ * This function takes an array of values and casts it to a tuple type where the first element is of the same type as the array, and the rest are of the same type as well.
+ */
+const zodEnum = <T>(arr: T[]): [T, ...T[]] => arr as [T, ...T[]];
+
+/**
+ * Create Zod enum schema from uppercase enum values
+ */
+const auditActionSchema = z.enum(zodEnum(upperCaseAuditActions));
```

Comment:
This is all very round about way of defining an enum with a zod schema. Just make the original declaration of AUDIT_ACTION using a zod.enum. This will also make the type assertion in the return statement below unnecessary.

Reply:
Used another approach to avoid impact other files that are using original AUDIT_ACTION. 
Assigned a Zod nativeEnum with AUDIT_ACTION and transform it to uppercase values maintaining the type.